src/services/NotificationService.js

```diff
@@ -28,26 +29,146 @@ class NotificationService {
+      tripRequestId,
+      message
+    });
+
     const data = {
+      notificationId: result.dataValues.id,
       tripRequestId,
-      status: statusName,
       reason,
       updatedAt
     };
+    notificationEvents('approve_reject_notification', { message, data });
+  }
 
-    await CommonQueries.create(notifications, {
-      userId,
-      tripRequestId,
+  /**
+   * generate notification
+   * @static
+   * @param {object} dataNotification pass object to save in Table
+   * @memberof NotificationService
+   * @returns {object} data
+   */
+  static async saveNotification(dataNotification) {
+    const data = await CommonQueries.create(notifications, dataNotification);
+    return data;
+  }
+
+  /**
+   * Send notification to manager when trip edited
+   * @static
+   * @param {object} req get user information
+   * @memberof NotificationService
+   * @returns {object} data
+   */
+  static async editedTripNotification(req) {
+    const { params, user } = req;
+    const managerById = {
+      where: { userId: user.id }
+    };
+
+    const userTripsRequestsObject = {
+      where: { id: params.tripRequestId }
+    };
+
+    const tripRequestInfo = await CommonQueries.findOne(tripRequests, userTripsRequestsObject);
+    const userProfileInfo = await CommonQueries.findOne(userProfile, managerById);
+    const { managerId } = userProfileInfo.dataValues;
+    const { updatedAt } = tripRequestInfo.dataValues;
+    const message = `Trip request no ${params.tripRequestId} has been edited`;
+
+    const result = await NotificationService.saveNotification({
+      userId: managerId,
+      tripRequestId: params.tripRequestId,
       message
     });
-    eventEmitters.emit('notification_message', JSON.stringify({ message, data }));
+
+    const data = {
+      notificationId: result.dataValues.id,
+      tripRequestId: params.tripRequestId,
+      updatedAt
+    };
+    notificationEvents('edit_trip_notification', { message, data });
+  }
+
+  /**
+   * Send notification to a user or manager when post added
+   * @static
+   * @param {object} req get user information
+   * @memberof NotificationService
+   * @returns {object} data
+   */
+  static async addCommentNotification(req) {
+    const { user, params } = req;
+    const tripRequestInfoObject = {
+      where: {
+        id: params.tripRequestId
+      }
+    };
+
+    const tripRequestInfo = await CommonQueries.findOne(tripRequests, tripRequestInfoObject);
+    const { userId } = tripRequestInfo;
+    const lineManagerIdObject = {
+      where: {
+        userId
+      }
+    };
+    const userProfileInfo = await CommonQueries.findOne(userProfile, lineManagerIdObject);
+    const { managerId } = userProfileInfo.dataValues;
+    const message = `The Trip request no ${params.tripRequestId} has been commented on`;
+    let receiverId;
+
+    if (userId === user.id) {
+      receiverId = managerId; // is a manage
+    } else {
+      receiverId = userId; // is a user
+    }
+
```

Comment:
@NiyoEric How do you know who you are going to send the notification?

Reply:
If user id saved in trips table equal to user id from token, that means a user adds a comment. The notification should be sent to the manager. Vice versa.

Reply:
Okay, It makes sense.